src/cow.rs

```diff
@@ -0,0 +1,85 @@
+#![cfg(feature = "alloc")]
+
+extern crate alloc;
+
+use crate::len_type::LenType;
+use crate::string::StringView;
+use crate::String;
+use core::borrow::Borrow;
+
+/// A clone-on-write (COW) string type for heapless.
+///
+/// `Cow` can either **borrow** a `StringView` or **own** a `String`.
+/// This allows efficient handling of strings that may be either temporary
+/// references or fully owned data.
+///
+/// # Type Parameters
+///
+/// - `N`: The inline buffer size for owned strings.
+/// - `LenT`: The integer type used for length tracking (must implement [`LenType`]).
+
+/// A clone-on-write string type for heapless
+#[derive(Debug)]
+pub enum Cow<'a, const N: usize, LenT: LenType = usize> {
+    /// A borrowed view of a string.
+    Borrowed(&'a StringView<LenT>),
+
+    /// An owned string with inline storage of size `N`.
+    Owned(String<N, LenT>),
```

Comment:
Since this is going to be string-specific, it's probably best to take advantage of that fact and also support another variant that never deep-clones (even on `to_owned`): `Static(&'static StringView<LenT>)`. This is similar to [what we added in zvariant](https://github.com/dbus2/zbus/blob/main/zvariant/src/str.rs#L34).